init/cmdline.go

```diff
@@ -127,23 +127,29 @@ func parseParams(params string) error {
 				return fmt.Errorf("invalid UUID %s %v", parts[0], err)
 			}
 
-			dev := luksMapping{
-				ref:  &deviceRef{refFsUUID, uuid},
-				name: parts[1],
-			}
-			luksMappings = append(luksMappings, dev)
+			var o = findOrCreateLuksMapping(uuid)
+			o.name = parts[1]
 		case "rd.luks.uuid":
 			stripped := stripQuotes(value)
-			u, err := parseUUID(stripped)
+			uuid, err := parseUUID(stripped)
 			if err != nil {
 				return fmt.Errorf("invalid UUID %s in rd.luks.uuid boot param: %v", value, err)
 			}
 
-			dev := luksMapping{
-				ref:  &deviceRef{refFsUUID, u},
-				name: "luks-" + stripped,
+			findOrCreateLuksMapping(uuid)
+		case "rd.luks.key":
```

Comment:
This parameter needs to be documented in the `docs/`

Reply:
Will work on that, also examining what options are available to fetch from another device. Stubbed out some code for working on it, will investigate more first to see what the challenges are for that portion.

Reply:
"fetch from another device" (either EFI or mountable USB) is a large task that will probably require some refactoring. I suggest leaving it for another pull request.